packages/expo-router/src/layouts/StackClient.tsx

```diff
@@ -29,8 +30,45 @@ type GetId = NonNullable<RouterConfigOptions['routeGetIdList'][string]>;
 
 const NativeStackNavigator = createNativeStackNavigator().Navigator;
 
+/**
+ * We extend NativeStackNavigationOptions with our custom props
+ * to allow for several extra props to be used on web, like modalWidth
+ */
+export type ExtendedStackNavigationOptions = NativeStackNavigationOptions & {
+  /**
+   * Override the width of the modal (px or percentage). Only applies on web platform.
+   * @platform web
+   */
+  modalWidth?: number | string;
+  /**
+   * Override the height of the modal (px or percentage). Applies on web desktop.
+   * @platform web
+   */
+  modalHeight?: number | string;
+  /**
+   * Minimum height of the desktop modal (px or percentage). Overrides the default 640px clamp.
+   * @platform web
+   */
+  modalMinHeight?: number | string;
+  /**
+   * Minimum width of the desktop modal (px or percentage). Overrides the default 580px.
+   * @platform web
+   */
+  modalMinWidth?: number | string;
+  /**
+   * Override the border of the desktop modal (any valid CSS border value, e.g. '1px solid #ccc' or 'none').
+   * @platform web
+   */
+  modalBorder?: string;
+  /**
+   * Override the overlay background color (any valid CSS color or rgba/hsla value).
+   * @platform web
+   */
+  modalOverlayBackground?: string;
```

Comment:
maybe we can group them in some option like `modalStyle`?

I'm not really sure what is the best approach here

Reply:
I agree. Grouped them in webModalStyle and when from prefixing with "modal" to border, widht etc, as it lives under the new key.